sae.c

```diff
@@ -68,14 +68,21 @@
 /*
  * the functions H() and CN()
  */
-#define H_Init(ctx,x,l) HMAC_Init((ctx), (x), (l), EVP_sha256())
+#define H_Init(ctx,x,l) HMAC_Init_ex((ctx), (x), (l), EVP_sha256(), NULL)
 #define H_Update(ctx,x,l) HMAC_Update((ctx),(x),(l))
 #define H_Final(ctx,x) HMAC_Final((ctx), (x), &function_mdlen)
 
-#define CN_Init(ctx,x,l) HMAC_Init((ctx), (x), (l), EVP_sha256())
+#define CN_Init(ctx,x,l) HMAC_Init_ex((ctx), (x), (l), EVP_sha256(), NULL)
 #define CN_Update(ctx,x,l) HMAC_Update((ctx),(x),(l))
 #define CN_Final(ctx,x) HMAC_Final((ctx), (x), &function_mdlen)
 
+/* Pre 1.1.0 compatibility macros */
+#ifndef OPENSSL_API_COMPAT
+#define HMAC_CTX_new() calloc(sizeof(HMAC_CTX), 1)
+#define HMAC_CTX_free(ctx) { HMAC_CTX_cleanup(ctx); free(ctx); ctx = NULL; }
```

Comment:
not sure about openssl's semantics, but usually free() can be called on null which can make error paths a little easier to write, so I would suggest wrapping it all in `if (ctx) {` .  Also static function might be cleaner than macros.

Reply:
will change